core/store/migrations/0016_pipeline_task_run_dot_id.go

```diff
@@ -0,0 +1,48 @@
+package migrations
+
+import (
+	"github.com/go-gormigrate/gormigrate/v2"
+	"gorm.io/gorm"
+)
+
+const (
+	up16 = `
+ALTER TABLE pipeline_task_runs ADD COLUMN dot_id text; 
+UPDATE pipeline_task_runs SET dot_id = ts.dot_id FROM pipeline_task_specs ts WHERE ts.id = pipeline_task_runs.pipeline_task_spec_id;
+ALTER TABLE pipeline_task_runs ALTER COLUMN dot_id SET NOT NULL; 
+
+ALTER TABLE pipeline_task_runs DROP COLUMN pipeline_task_spec_id;
+DROP TABLE pipeline_task_specs;
+
+CREATE UNIQUE INDEX ON pipeline_task_runs(pipeline_run_id, dot_id);
```

Comment:
We can probably remove `idx_pipeline_task_runs_optimise_find_results` now since this one covers it.